src/Models/Menu.php

```diff
@@ -30,4 +30,10 @@ public function items()
     {
         return $this->hasMany(MenuItem::class, 'menu_id');
     }
+
+    // Return only direct descendants
+    public function children()
+    {
+        return $this->items()->whereNull('parent_id');
```

Comment:
This does not seem right. Can you please explain what is does?

Reply:
It's getting all of the MenuItems which belong to the Menu (the "items") relationship, but only the ones that are top level MenuItems (e.g they don't have a parent MenuItem)